src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+        }
+        return out;
+    }
+
+    /**
+     * <p>
+     * Converts binary (represented as boolean array) into a byte using the default (little
+     * endian, Lsb0) byte and bit ordering.
+     * </p>
+     *
+     * @param src the binary to convert
+     * @param srcPos the position in {@code src}, in boolean unit, from where to start the
+     *            conversion
+     * @param dstInit initial value of the destination byte
+     * @param dstPos the position of the lsb, in bits, in the result byte
+     * @param nBools the number of booleans to convert
+     * @return a byte containing the selected bits
+     * @throws NullPointerException if {@code src} is {@code null}
+     * @throws IllegalArgumentException if {@code nBools-1+dstPos >= 8}
+     * @throws ArrayIndexOutOfBoundsException if {@code srcPos + nBools > src.length}
+     */
+    public static byte binaryToByte(final boolean[] src, final int srcPos, final byte dstInit, final int dstPos,
+            final int nBools) {
+        if (src.length == 0 && srcPos == 0 || 0 == nBools) {
+            return dstInit;
+        }
+        if (nBools - 1 + dstPos >= 8) {
+            throw new IllegalArgumentException("nBools-1+dstPos is greater or equal to than 8");
+        }
+        byte out = dstInit;
+        for (int i = 0; i < nBools; i++) {
+            final int shift = i + dstPos;
+            final int bits = (src[i + srcPos] ? 1 : 0) << shift;
+            final int mask = 0x1 << shift;
+            out = (byte) ((out & ~mask) | bits);
+        }
+        return out;
+    }
+
+    /**
+     * <p>
+     * Converts a long into an array of int using the default (little endian, Lsb0) byte and bit
+     * ordering.
+     * </p>
+     *
+     * @param src the long to convert
+     * @param srcPos the position in {@code src}, in bits, from where to start the conversion
+     * @param dst the destination array
+     * @param dstPos the position in {@code dst} where to copy the result
+     * @param nInts the number of ints to copy to {@code dst}, must be smaller or equal to the
+     *            width of the input (from srcPos to msb)
+     * @return {@code dst}
+     * @throws NullPointerException if {@code dst} is {@code null} and {@code nInts > 0}
+     * @throws IllegalArgumentException if {@code (nInts-1)*32+srcPos >= 64}
+     * @throws ArrayIndexOutOfBoundsException if {@code dstPos + nInts > dst.length}
+     */
+    public static int[] longToIntArray(final long src, final int srcPos, final int[] dst, final int dstPos,
+            final int nInts) {
+        if (0 == nInts) {
+            return dst;
+        }
+        if ((nInts - 1) * 32 + srcPos >= 64) {
+            throw new IllegalArgumentException("(nInts-1)*32+srcPos is greater or equal to than 64");
+        }
+        for (int i = 0; i < nInts; i++) {
+            final int shift = i * 32 + srcPos;
+            dst[dstPos + i] = (int) (0xffffffff & (src >> shift));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).